flow/operations.py

```diff
@@ -90,6 +90,15 @@ def hello(job):
         def hello_cmd(job):
             return 'trap "cd `pwd`" EXIT && cd {} && echo "hello {job}"'.format(job.ws)
     """
+    signature = inspect.signature(func)
+    if len(signature.parameters) > 1:
+        raise ValueError("Only a single 'positional argument' is allowed "
+                         "when @with_job decorator is used")
+    for v in signature.parameters.values():
+        if v.kind is inspect.Parameter.VAR_POSITIONAL:
+            raise ValueError("Only a single 'positional argument' is allowed "
+                             "when @with_job decorator is used")
+
```

Comment:
This doesn't work for the case `foo(*jobs)` you need to check the single parameter's `kind` attribute. If it is a variable number of positional arguments `_ParameterKind.VAR_POSITIONAL`, then we should raise a `ValueError` as well.

Reply:
```
@with_job
@FlowProject.operation
def op_1(*jobs):
    print('op_1')
```
@b-butler  Executing this raises an error in my case.